tasks/Logger.cpp

```diff
@@ -59,10 +63,16 @@ bool Logger::startHook()
     if (_file.value().empty())
         return false;
 
-    if(boost::filesystem::exists(_file.value()) && !_overwrite_existing_files.get())
+    if(boost::filesystem::exists(_file.value()))
     {
-        log(Error) << "File " << _file.value() << " already exists." << endlog();
-        return false;
+        if(!_overwrite_existing_files.get()){
+            // increment the log file number to store it in a new file
+            vector<string> strs;
+            boost::split(strs, _file.value(), boost::is_any_of("."));
+            int nr = stoi(strs[strs.size()-2]);
```

Comment:
There are a lot of assumptions on the name of the log file here... And on the state of the component (that _file + 1 is free)

Generally speaking, I don't think it's a good idea to "just increment" the value in `_file`. It's too fragile. Rather, add an `auto_increment` property (initially set to false) and when this property is set, require _file to contain a sprintf construct (e.g. `%03i`). In the startHook, look for the first number starting at zero which does not exist yet.

Verifying that _file does contain the sprintf construct can be done by comparing the value of _file and the output value of sprintf (note that there's probably a boost equivalent of sprintf)

Finally, properties are not designed to be set by components. If you want to report on the selected file path, add an attribute with e.g. `actual_file_path` and set it with the real path.